src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User login(String email, String password)
+            throws AuthenticationException {
+        Optional<User> userOptional = userDao.findByEmail(email);
+        if (userOptional.isEmpty() || !userOptional.get().getPassword().equals(password)) {
```

Comment:
we have hashed password in db, and not hashed password as input parameter, could they be equal?